SCIM: Add service provider config and docs (#47437)
sourcegraph/sourcegraph: doc/admin/scim.md

# SCIM

> **⚠️ DISCLAIMER: SCIM SUPPORT IS UNDER DEVELOPMENT. THE ENDPOINT DOES NOT COMPLY WITH THE SCIM STANDARD YET AND IT ISN’T READY FOR PRODUCTION USE.**

Reviewer: ```suggestion
**⚠️ DISCLAIMER: SCIM SUPPORT IS UNDER DEVELOPMENT. THE ENDPOINT DOES NOT COMPLY WITH THE SCIM STANDARD YET AND IT IS NOT READY FOR PRODUCTION USE.**
```
nit :)

Author: I think my version is more in line with our style and mechanics guide: https://handbook.sourcegraph.com/company-info-and-process/communication/content_guidelines/style_and_mechanics/#contractions
WDYT?

Reviewer: The sentence just sounds wrong to me when I say it out loud 🙈 

I think it's because the *NOT* sounds important to me, so contracting it away feels counter productive. Similarly, earlier in the sentence you say `DOES NOT` instead of `DOESN'T`, and I think it's for the same reason: it feels important to highlight that it is NOT ready for production.

What about saying `AND IT'S NOT READY FOR PRODUCTION USE` instead, moving the contraction to the `IT + IS` rather than the `IS + NOT` ?

> SCIM is a standard for provisioning users and groups in an organization. It is supported by many IdP (identity providers) such as Okta, OneLogin, and Azure Active Directory.

Reviewer: ```suggestion
SCIM (System for Cross-domain Identity Management) is a standard for provisioning users and groups in an organization. It is supported by many IdP (identity providers) such as Okta, OneLogin, and Azure Active Directory.
```

Sourcegraph supports SCIM for provisioning and de-provisioning _users_, but not groups at this time.

> You can use any identity provider that supports SCIM, but we’ve only tested the endpoint with Okta and Azure Active Directory.

Reviewer: ```suggestion
You can use any IdP that supports SCIM, but we’ve only tested the endpoint with Okta and Azure Active Directory.
```

Think I'd prefer to be consistent with term usage here after definitions 🤔

## How to use

> To use SCIM, you must have an existing identity provider (IdP) configured with your Sourcegraph instance.

Reviewer: ```suggestion
To use SCIM, you must have an existing IdP configured with your Sourcegraph instance.
```

Has already been defined :) unless you want to redefine it in each section. Not sure what the best practice is here.
